server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -471,13 +260,41 @@ protected Query doToQuery(QueryShardContext context) {
         return query;
     }
 
+    public static SpatialArgs getArgs(Geometry shape, ShapeRelation relation) {
+        switch (relation) {
+            case DISJOINT:
+                return new SpatialArgs(SpatialOperation.IsDisjointTo, buildS4J(shape));
+            case INTERSECTS:
+                return new SpatialArgs(SpatialOperation.Intersects, buildS4J(shape));
+            case WITHIN:
+                return new SpatialArgs(SpatialOperation.IsWithin, buildS4J(shape));
+            case CONTAINS:
+                return new SpatialArgs(SpatialOperation.Contains, buildS4J(shape));
+            default:
+                throw new IllegalArgumentException("invalid relation [" + relation + "]");
+        }
+    }
+
+    /**
+     * Builds JTS shape from a geometry
+     *
+     * This method is needed to handle legacy indices and will be removed when we no longer need to build JTS shapes
+     */
+    private static Shape buildS4J(Geometry geometry) {
+        return geometryToShapeBuilder(geometry).buildS4J();
+    }
+
     private Query getVectorQuery(QueryShardContext context, Geometry queryShape) {
```

Comment:
I think in the next iteration I am going to move this into `AbstractGeometryIndexer` and make it obtainable from `QueryShardContext`.

Reply:
Do you want to also `@deprecate` it in 8.0 (master) in advance of its removal in 9.0?

Reply:
Oh... oops.. Sorry, you were referring to `getVectorQuery`.